### PR TITLE
fix(monitor-v2): wait for logger

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

Bots in monitor-v2 package did not properly wait for logger to flush.

**Summary**

Adds waiting for logger to flush in monitor-v2 package.

**Details**

Existing 5 second delay was sufficient for logger to deliver on all transports for bots in monitor-v2 package. With the introduction of Discord Ticket Tool Transport that highly depends on back-off due to rate limiting this might not be sufficient and we need feedback from transports to check if process can safely exit.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/monitor-v2/src/bot-oo-v3/index.ts
Comment (mrice32) [inline packages/monitor-v2/src/bot-oo-v3/index.ts:36]: We have to be very careful with this change because if any transport sets it to false and never sets it back, it could cause bots to hang, causing the hub to time out. Just looking at the existing [discord transport](https://github.com/UMAprotocol/protocol/blob/d314228f5bae1e7ba03eca7cc98fe20c6fb7d7d1/packages/financial-templates-lib/src/logger/DiscordTransport.ts#L106-L117) (which I think is the only bot to touch isFlushed) can end up in an infinite recursive error loop. We might want to make sure that it terminates after some finite number of calls or times out before adding this to all the bots.

Thoughts?

**Review by mrice32 — COMMENTED** [review]
Looks good! Just one question

File: packages/monitor-v2/src/bot-oo-v3/index.ts
Comment (Reinis-FRP) [reply packages/monitor-v2/src/bot-oo-v3/index.ts:36]: We already use waitForLogger in the monitors package that covers all other versions of OO, except for OOv3 and monitors package still constitutes for most of alerts.

But I agree its better to have some control not allowing this to hang. One solution would be to have logger timeout (e.g. defaults to 5 minutes) that allows to return early from waitForLogger even if not all transports are flushed. Better to do this in #4607 as its already changing its behavior. And after #4609 we could also send alert to the dedicated log error channel that some transports had failed to flush.

Comment (Reinis-FRP) [reply packages/monitor-v2/src/bot-oo-v3/index.ts:36]: Added logger flush timeout in #4607

Comment (mrice32) [reply packages/monitor-v2/src/bot-oo-v3/index.ts:36]: LGTM once #4607 is merged!

**Review by mrice32 — APPROVED** [review]

**Pull request merged**
